### PR TITLE
Improvements related to handling of dead kernels

### PR DESCRIPTION
**Pull request by DonJayamanne**
For #8060

**Summary:**
* If kernel dies, display an error message notifying user its dead (bottom right)
	* We do this only for raw kernels, as jupyter automatically restarts kernels 
	* Automatic restarts of raw kernels will be done in a separate PR
* If user attempts to run a cell against a dead kernel, they get a modal dialog
	* With options to restart kernel or view output
	* Today - user gets an error message `Session cannot generate requests` error
* When a cell dies amid exrecution of a notebook, we cancel all remaining executions (no point)
	* This way we do'nt end with error mesages (session disposed) in the output   	 
* When kernels die, ensure we display a meaningful error message where possible
	* With ability to view the output logs

**TODO:** 
Will add tests, just want to get the stuff reviewed

### CODE REVIEW

**Review by rchiodo — APPROVED** [review]

File: src/client/datascience/jupyter/kernels/kernel.ts [hunk 1]
```diff
@@ -329,9 +351,17 @@ export class Kernel implements IKernel {
                             this.notebookDocument,
                             placeholderCellPromise
                         );
+                        // if (!process.env.XYZ1234) {
```
Comment (IanMatthewHuff) [inline src/client/datascience/jupyter/kernels/kernel.ts:354]: ?

File: package.nls.json [hunk 2]
```diff
@@ -433,9 +435,11 @@
     "DataScience.selectDifferentJupyterInterpreter": "Change Interpreter",
     "DataScience.selectJupyterInterpreter": "Select an Interpreter to start Jupyter",
     "DataScience.switchingKernelProgress": "Switching kernel to '{0}'",
-    "DataScience.sessionStartFailedWithKernel": "Failed to start a session for the Kernel '{0}'. \nView Jupyter [log](command:{1}) for further details.",
+    "DataScience.sessionStartFailedWithKernel": "Failed to start the Kernel '{0}'. \nView Jupyter [log](command:{1}) for further details.",
```
Comment (IanMatthewHuff) [inline package.nls.json:438]: Looks like for sentences we use kernel not Kernel in other strings in this file.

**Review by IanMatthewHuff — APPROVED** [review]

File: src/client/datascience/jupyter/kernels/kernel.ts (hunk 1 above)
Comment (DonJayamanne) [reply src/client/datascience/jupyter/kernels/kernel.ts:354]: I've left that for testing, will remove.
I thought my comments were on github, hmm

Comment (DonJayamanne) [reply src/client/datascience/jupyter/kernels/kernel.ts:355]: Fixed

**Pull request merged**